SECEdgar/filings/base.py

```diff
@@ -145,17 +146,27 @@ def _make_dir(self, directory):
     def _sanitize_path(dir):
         return os.path.expanduser(dir)
 
-    def save(self, directory):
+    def save(self, directory, extract=False, extract_dir=None):
         """Save files in specified directory.
 
         Args:
             directory (str): Path to directory where files should be saved.
+            extract (bool): whether to extract the documents from the downloaded .txt files
```

Comment:
I don't think extractor responsibility will go to the `Filing` class. Filing job is to just process individual filings. The extractor can live it on its own. Meaning, we can pass the Filing object to the extractor and it will process the file.